repl/src/main/scala/com/cloudera/livy/repl/PythonInterpreter.scala

```diff
@@ -205,13 +207,37 @@ private class PythonInterpreter(process: Process, gatewayServer: GatewayServer,
     }
   }
 
-  @tailrec
   final override protected def waitUntilReady(): Unit = {
+    waitForGateway()
+    // LIVY-294, need to check whether HiveContext can work properly, fallback to SQLContext if
+    // HiveContext can not be initialized successfully. Only for spark-1.
+    val code =
+      """
+        |import py4j
+        |from pyspark.sql import SQLContext
+        |try:
+        |  sqlContext.tables()
+        |except py4j.protocol.Py4JError:
+        |    sqlContext = SQLContext(sc)
```

Comment:
Also we should check whether `LIVY_TEST` is configured, this should not be created if running in unit test.